serial.ts

```diff
@@ -186,79 +209,86 @@ class SerialPort {
    * @return {boolean} A boolean that reflects whether the stop bits setting is
    * valid
    */
-  isValidStopBits_(stopbits) {
-    return acceptableStopBits.includes(stopbits);
+  private isValidStopBits(stopbits: number) {
+    return kAcceptableStopBits.includes(stopbits);
   }
 
   /**
    * Checks the parity for validity
-   * @param {number} parity the parity to check
+   * @param {string} parity the parity to check
    * @return {boolean} A boolean that reflects whether the parity is valid
    */
-  isValidParity_(parity) {
-    return acceptableParity.includes(parity);
+  private isValidParity(parity: ParityType) {
+    return kAcceptableParity.includes(parity);
   }
 
   /**
    * The function called by the writable stream upon creation
-   * @param {number} controller The stream controller
+   * @param {WritableStreamDefaultController} controller The stream controller
    * @return {Promise} A Promise that is to be resolved whe this instance is
    * ready to use the writablestream
    */
-  writeStart_(controller) {
-    return new Promise((resolve, reject) => {
-      if (this.device_) {
-        resolve();
-      }
-    });
+  private async writeStart(controller: WritableStreamDefaultController) {
   }
 
   /**
    * The function called by the readable stream upon creation
    * @param {number} controller The stream controller
    */
-  readStart_(controller) {
-    const readLoop = () => {
-      this.device_
-          .transferIn(this.getDirectionEndpoint_('in').endpointNumber, 64)
-          .then(
-              (result) => {
-                controller.enqueue(result.data);
-                readLoop();
-              },
-              (error) => {
-                controller.error(error.toString());
-              });
-    };
-    readLoop();
+  private async readStart_(controller: ReadableStreamDefaultController) {
```

Comment:
I am on board with private functions not having underscores.  However, you forgot one.